include/alloc/heapalloc.h

```diff
@@ -5,13 +5,31 @@
 
 #include "stdlib/stdtypes.h"
 
+/** Represents a node of metadata that maps to the corresponding element in the heap struct's memory page */
+struct Node {
```

Comment:
You should move the declaration of this struct into the c file instead of having it in the h file